src/Microsoft.AspNetCore.Authentication/PersistentAuthenticationHandler.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Antiforgery;
+using Microsoft.AspNetCore.Builder;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    public abstract class PersistentAuthenticationHandler<TOptions> : AuthenticationHandler<TOptions>
+        where TOptions : AuthenticationOptions
+    {
+        public PersistentAuthenticationHandler(IAntiforgery antiforgery)
+        {
+            if (antiforgery == null)
+            {
+                throw new ArgumentNullException(nameof(antiforgery));
+            }
+
+            Antiforgery = antiforgery;
+        }
+
+        protected IAntiforgery Antiforgery { get; }
+        
+        protected async Task<bool> ValidateAntiforgeryAsync()
+        {
+            if (RequiresAntiforgery())
+            {
+                return await Antiforgery.IsRequestValidAsync(Context);
+            }
+
+            return true;
+        }
+
+        protected virtual bool RequiresAntiforgery()
+        {
+            var method = Context.Request.Method;
```

Comment:
Yeah I think you're right. This is starting to feel like these pieces need to live there.